### DISCUSSION TITLE
Is there a way to add a marker in any of the simulators?

### DISCUSSION THREAD

**Discussion by slagernate** [cocotb/cocotb#3127] [Proposals]
Perhaps there's a better way of accomplishing this, but basically I'd like to know when something is triggering without having to print out the time and manually zooming into it. It would be helpful to be able to add a marker in the simulator (multiple would be ideal). Currently using Modelsim but I am curious if this is possible in _any_ simulator. Not familiar with the gpi interface so figured I'd ask the community.

**Comment by themperek** [comment]
Would it be possible to trigger some (artificial) timing check to fail? This shows some kind of marker if i am not wrong.

**Reply by slagernate** [reply]
which artificial timing check? I'm not aware of any such check which would produce a marker in the simulator.

**Reply by themperek** [reply]
Add timing check and trigger a timing error:
```
$width(negedge a, 1);
```

Use:
```
vsim -msgmode both
```

But adding a fake fale signal also works.

**Comment by slagernate** [comment]
Was actually able to insert my own faux markers by introducing a fast clock (non-periodic with my design's clock), and a 1 bit signal (to act as a marker). At the negative edge of my fast clock I am setting the 1 bit marker signal to 0. In my cocotb I can set the value to 1 at virtually any time. This will show up as a small pulse, as the fast clock will soon thereafter reset the marker signal to 0.
